Content.Server/Disease/DiseaseSystem.cs

```diff
@@ -0,0 +1,420 @@
+        private void InteractWithDiseased(EntityUid diseased, EntityUid target)
+        {
+            if (!TryComp<DiseaseCarrierComponent>(target, out var carrier))
+                return;
+
+            var disease = _random.Pick(Comp<DiseaseCarrierComponent>(diseased).Diseases);
+            if (disease != null)
+                TryInfect(carrier, disease, 0.4f);
+        }
+        /// <summary>
+        /// Adds a disease to a target
+        /// if it's not already in their current
+        /// or past diseases. If you want this
+        /// to not be guaranteed you are looking
+        /// for TryInfect.
+        /// </summary>
+        public void TryAddDisease(DiseaseCarrierComponent? target, DiseasePrototype? addedDisease, string? diseaseName = null, EntityUid host = default!)
+        {
+            if (diseaseName != null && _prototypeManager.TryIndex(diseaseName, out DiseasePrototype? diseaseProto))
+                addedDisease = diseaseProto;
+
+            if (host != default!)
+                target = Comp<DiseaseCarrierComponent>(host);
+
+            if (target != null)
+            {
+                foreach (var disease in target.AllDiseases)
+                    if (disease.Name == addedDisease?.Name) //Name because of the way protoypes work
+                        return;
+            var freshDisease = _serializationManager.CreateCopy(addedDisease) ?? default!;
+            target.Diseases.Add(freshDisease);
+            AddQueue.Enqueue(target.Owner);
+            }
+        }
+        /// <summary>
+        /// Pits the infection chance against the
+        /// person's disease resistance and
+        /// rolls the dice to see if they get
+        /// the disease.
+        /// </summary>
+        public void TryInfect(DiseaseCarrierComponent carrier, DiseasePrototype? disease, float chance = 0.7f)
+        {
+            if(disease == null || !disease.Infectious)
+                return;
+            var infectionChance = chance - carrier.DiseaseResist;
+            if (infectionChance <= 0)
+                return;
+            if (_random.Prob(infectionChance))
+                TryAddDisease(carrier, disease);
+        }
+        /// <summary>
+        /// Plays a sneeze/cough popup if applicable
+        /// and then tries to infect anyone in range
+        /// if the snougher is not wearing a mask.
+        /// </summary>
+        public void SneezeCough(EntityUid uid, DiseasePrototype? disease, SneezeCoughType Snough, float infectionChance = 0.3f)
+        {
+            IngestionBlockerComponent blocker;
+
+            var xform = Comp<TransformComponent>(uid);
+
+            if (Snough == SneezeCoughType.Sneeze)
+                _popupSystem.PopupEntity(Loc.GetString("disease-sneeze", ("person", uid)), uid, Filter.Pvs(uid));
+            else if (Snough == SneezeCoughType.Cough)
+                _popupSystem.PopupEntity(Loc.GetString("disease-cough", ("person", uid)), uid, Filter.Pvs(uid));
```

Comment:
this might as well just be a bool instead of an enum if this is the only practical difference, then rename all the 'SneezeCough' stuff to just 'AirTransmit' or something